sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs

```diff
@@ -2954,7 +2954,7 @@ public static partial class SearchModelFactory
         public static Azure.Search.Documents.Indexes.Models.CognitiveServicesAccount CognitiveServicesAccount(string oDataType, string description) { throw null; }
         public static Azure.Search.Documents.Indexes.Models.DataChangeDetectionPolicy DataChangeDetectionPolicy(string oDataType) { throw null; }
         public static Azure.Search.Documents.Indexes.Models.DataDeletionDetectionPolicy DataDeletionDetectionPolicy(string oDataType) { throw null; }
-        public static Azure.Search.Documents.Models.FacetResult FacetResult(long? count, System.Collections.Generic.IReadOnlyDictionary<string, object> additionalProperties) { throw null; }
+        public static Azure.Search.Documents.Models.FacetResult FacetResult(long? count = default(long?), System.Collections.Generic.IReadOnlyDictionary<string, object> additionalProperties = null) { throw null; }
```

Comment:
Is this not considered a breaking change?

Reply:
Since model factory is used only for mocking to enable non-live testing of service clients by customers, I'm not sure, if this is a breaking change.

Reply:
I don't think it's a breaking change, we are just providing default values for the all the parameters.

Reply:
I don't think it will break customer's existing code, since we are just providing default values for all the parameters.

Also, we are not changing public client. We are just updating model factory which is used for mocking to enable non-live testing of service clients by customers.

Reply:
@tg-msft could you please confirm if this is considered as a breaking change or not?

Reply:
I couldn't come up with a case where existing code would fail to resolve the parameters or bind to a different overload.   @pallavit - was there a scenario that you had in mind?

Reply:
The compiler only includes the default value at the call site, so

```C#
Test();
void Test(int x = 42) { }
```

compiles into 

```diff
.method private hidebysig static void '<Main>$' (string[] args) cil managed 
{
    .maxstack 8
    .entrypoint
    ldc.i4.s 42
    call void '<Program>$'::'<<Main>$>g__Test|0_0'(int32)
    ret
}

.method assembly hidebysig static void '<<Main>$>g__Test|0_0' (int32 x) cil managed 
{
    .maxstack 8
    ret
}
```

That luckily means we don't have to worry about binary breaking changes!  It is possible for tweaking default values to introduce source breaking changes though.  For example, doing something like adding an overload to switch to an options bag:

```diff
- void Test(int x = 42, string y = "hello")
+ void Test(int x, string y)
+ void Test(TestOptions options = default)
```

would cause a compile error like `error CS1503: Argument 1: cannot convert from 'int' to 'TestOptions'` for anyone using `Test(21)` today.

We try really hard to avoid those, but sometimes let them through.  Default values in C# unfortunately version worse than most other language features.  We've had a lot of discussion with the .NET team about this and their view is we're using overloads incorrectly/too aggressively.  :smile:

All that said, this particular example isn't going to break anyone.

Reply:
Thank you everyone! Yes my conclusion based on all the breaking change conversations about optional arguments was that everything regarding optional arguments is breaking :P. 

Apologies for replying on a merged thread but I was curious what we do in these scenarios :)